plugins/modules/zos_job_submit.py

```diff
@@ -745,57 +749,91 @@ def run_module():
     except SubmitJCLError as e:
         module.fail_json(msg=repr(e), **result)
     if jobId is None or jobId == "":
-        result["job_id"] = jobId
+        result["job_id"] = "-blank-"
         module.fail_json(
             msg="JOB ID RETURNED IS None. PLEASE CHECK WHETHER THE JCL IS CORRECT.",
             **result
         )
 
     result["job_id"] = jobId
     duration = 0
-    if wait is True:
-        # calculate the job elapse time
+    if not wait:
+        wait_time_s = 10
```

Comment:
Although I don't disagree with change, this `wait_time_s=10` seems to now conflict and have changed our original behavior without any doc updates. Unless I read something wrong it looks like when wait is not set, it will always be 10 seconds thus never truly failing fast, we improved fail fast in one case and deactivated in another, unless I am reading the code incorrectly.

Doc in question:
```
wait

    Wait for the Job to finish and capture the output. Default is false.

    User can specify the wait time, see option wait_time_s.
    required: False
    type: bool
```

Reply:
I interpreted "Fail fast" to mean "fail when we have a failure message", like JCL ERROR.  

Setting wait_time_s to 10 means that at or after 10 seconds have elapsed (checked every 0.5 seconds), it will return with the 'long running job' message, because there has been no terminating response.  We have to set None/0 to something, or undefined wait would mean check once and call it a long running process.

Here's where I'm at a logical sticking point:
    wait:
     - Wait for the Job to finish and capture the output. Default is false.
If we want that to be  strictly true, we may need to change the 'long running job' to only reply if wait is true.  Even if we do, how we respond would depend on whether we received a terminating response.  That was actually my thinking, was the 10 seconds would be until termination or long-running-job... I forgot about the documentation above, so I do think we need to hash this out together.

Reply:
Let me shelve this comment and think about it more, as I think about it now, us polling infinitely on a long running job could promote a bad design pattern such that that playbook task never returns, better to have the user loop and react where they would set up their own poll with zos_job_query depending on what the result is.